setup/src/Magento/Setup/Model/ConfigGenerator.php

```diff
@@ -227,22 +238,35 @@ public function createModeConfig()
      */
     public function createCacheHostsConfig(array $data)
     {
-        $configData = new ConfigData(ConfigFilePool::APP_ENV);
+        $configData = $this->configDataFactory->create(ConfigFilePool::APP_ENV);
+
         if (isset($data[ConfigOptionsListConstants::INPUT_KEY_CACHE_HOSTS])) {
-            $hostData = explode(',', $data[ConfigOptionsListConstants::INPUT_KEY_CACHE_HOSTS]);
-            $hosts = [];
-            foreach ($hostData as $data) {
-                $dataArray = explode(':', trim($data));
-                $host = [];
-                $host['host'] = $dataArray[0];
-                if (isset($dataArray[1])) {
-                    $host['port'] = $dataArray[1];
-                }
-                $hosts[] = $host;
-            }
+            $hosts = explode(',', $data[ConfigOptionsListConstants::INPUT_KEY_CACHE_HOSTS]);
+            $hosts = array_map([$this, 'mapHostData'], $hosts);
             $configData->set(ConfigOptionsListConstants::CONFIG_PATH_CACHE_HOSTS, $hosts);
         }
+
         $configData->setOverrideWhenSave(true);
         return $configData;
     }
+
+    /**
+     * Splits host data to host and port and returns them as an assoc. array.
+     *
+     * @param string $hostData
+     *
+     * @return array
+     */
+    private function mapHostData(string $hostData) : array
```

Comment:
I guess we might need to use php mess dectector tags here so we ignore the error with UnusedPrivateMethod. The method is used in the array_map but the mess dectector does not pick this usage up.